dali/operators/generic/slice/slice_attr.h

```diff
@@ -273,24 +338,55 @@ class SliceAttr {
   bool normalized_anchor_, normalized_shape_;
   std::vector<int> axes_;
   TensorLayout axis_names_;
+  std::vector<CropWindowGenerator> crop_window_generators_;
+};
 
-  ArgValue<int, 1> start_;
-  ArgValue<float, 1> rel_start_;
-
-  ArgValue<int, 1> end_;
-  ArgValue<float, 1> rel_end_;
-
-  ArgValue<int, 1> shape_;
-  ArgValue<float, 1> rel_shape_;
+class SliceAttr {
+ public:
+  explicit inline SliceAttr(const OpSpec &spec)
+      : spec_(spec),
+        named_slice_attr_(spec),
+        pos_slice_attr_(spec) {
+  }
 
-  std::vector<CropWindowGenerator> crop_window_generators_;
+  template <typename Backend>
+  void ProcessArguments(const workspace_t<Backend> &ws) {
+    named_args_ = named_slice_attr_.template ProcessArguments<Backend>(ws);
+    if (named_args_) {
+      if (spec_.HasArgument("normalized_anchor") || spec_.HasArgument("normalized_shape")) {
+        DALI_WARN(
+            "Warning: ``normalized_anchor``/``normalized_shape`` is only relevant "
+            "when using positional slice arguments");
+      }
+      DALI_ENFORCE(ws.NumInput() <= 1,
+                  "Named arguments start/end/shape are not compatible with positional"
+                  " anchor and shape inputs");
+    } else {
+      if (ws.NumInput() != 3) {
```

Comment:
Perhaps the number of data inputs should be a member?
```suggestion
      DALI_ENFORCE(ws.NumInput() <= NumDataInputs(),
                  "Named arguments start/end/shape are not compatible with positional"
                  " anchor and shape inputs");
    } else {
      if (ws.NumInput() != NumDataInputs()) {
```
where
```
  int NumDataInputs() const {
    return spec_.GetSchema().MinNumInput();
  }
````